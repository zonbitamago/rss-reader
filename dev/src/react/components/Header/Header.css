.Header {
  background-color: #039be5;
  width: 100%;
  height: 3em;
  color: white;
  display: grid;
  grid-template-columns: 3em 1fr 6em;
}
.Header > div {
  height: inherit;
}
.Header-Left {
}
.Header-Center {
  display: flex;
  justify-content: center;
  align-items: center;
}
.Header-Center > div {
  margin: 0px 5px 0px 5px;
}
