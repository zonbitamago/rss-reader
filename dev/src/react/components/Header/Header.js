import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Header.css";
import Icon from "../Icon/Icon";
import Home from "@material-ui/icons/Home";
import RssFeed from "@material-ui/icons/RssFeed";
import Twitter from "react-icons/lib/ti/social-twitter";
import Typography from "material-ui/Typography";

class Header extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="Header">
        <div className="Header-Left">
          <Icon>
            <Home />
          </Icon>
        </div>
        <div className="Header-Center">
          <Icon>
            <RssFeed />
          </Icon>
          <Typography color="inherit" variant="subheading">
            +
          </Typography>
          <Icon>
            <Twitter />
          </Icon>
          <Typography color="inherit" variant="subheading">
            Feed
          </Typography>
        </div>
        <div className="Header-Right" />
      </div>
    );
  }
}

Header.propTypes = {};

Header.defaultProps = {};

export default Header;
