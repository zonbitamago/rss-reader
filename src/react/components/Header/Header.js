import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Header.css";
import Icon from "../Icon/Icon";
import Home from "@material-ui/icons/Home";
import RssFeed from "@material-ui/icons/RssFeed";
import Twitter from "react-icons/lib/ti/social-twitter";
import Typography from "@material-ui/core/Typography";
import $ from "jquery";

class Header extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="Header">
        <div className="Header-Left">
          <Icon>
            <Home
              onClick={() => {
                $(".main").animate({ scrollTop: 0 });
              }}
            />
          </Icon>
        </div>
        <div className="Header-Center">
          <Icon>
            <RssFeed />
          </Icon>
          <Typography color="inherit" variant="subheading">
            +
          </Typography>
          <Icon>
            <Twitter />
          </Icon>
          <Typography color="inherit" variant="subheading">
            Feed
          </Typography>
        </div>
        <div className="Header-Right">
          <div>update : {this.props.store.ItemStore.updateDate}</div>
        </div>
      </div>
    );
  }
}

Header.propTypes = {};

Header.defaultProps = {};

export default Header;
