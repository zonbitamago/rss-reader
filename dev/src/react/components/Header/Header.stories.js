import React from "react";
import { storiesOf } from "@storybook/react";

import Header from "./Header";

storiesOf("Organisms/Header", module).add("Header", () => <Header />);
