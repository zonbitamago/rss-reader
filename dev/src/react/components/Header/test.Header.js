import React from 'react'
import { shallow } from 'enzyme'

import Header from './Header'

describe('Header', () => {
  let component, props

  beforeEach(() => {
    props = {}
    component = shallow(<Header {...props} />)
  })

  it('should', () => {
    expect(component).toMatchSnapshot()
  })
})