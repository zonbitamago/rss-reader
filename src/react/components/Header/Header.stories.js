import React from "react";
import { storiesOf } from "@storybook/react";
import { store } from "../../../../__mocks__/storeSettings";

import Header from "./Header";

storiesOf("Organisms/Header", module).add("Header", () => (
  <Header store={store} />
));
