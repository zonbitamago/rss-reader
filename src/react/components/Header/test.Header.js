import React from "react";
import { shallow, mount } from "enzyme";
import $ from "jquery";
import { store } from "../../../../__mocks__/storeSettings";

import Header from "./Header";

jest.mock("jquery");

describe("Header", () => {
  let component, props;

  beforeEach(() => {
    props = { store: store };
    component = shallow(<Header {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });

  it("Home Icon Click", () => {
    $.mockImplementation(() => {
      return {
        animate: selector => {
          return;
        }
      };
    });
    var wrapper = mount(<Header {...props} />);

    wrapper.find(".Header-Left .Icon svg").simulate("click");
    expect($.mock.calls.length).toBe(1);
  });
});
