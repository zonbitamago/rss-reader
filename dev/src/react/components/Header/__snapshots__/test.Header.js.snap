// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Header-Left"
        >
          <Icon>
            <pure(Home) />
          </Icon>
        </div>,
        <div
          className="Header-Center"
        >
          <Icon>
            <pure(RssFeed) />
          </Icon>
          <WithStyles(Typography)
            color="inherit"
            variant="subheading"
          >
            +
          </WithStyles(Typography)>
          <Icon>
            <TiSocialTwitter />
          </Icon>
          <WithStyles(Typography)
            color="inherit"
            variant="subheading"
          >
            Feed
          </WithStyles(Typography)>
        </div>,
        <div
          className="Header-Right"
        />,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon>
            <pure(Home) />
          </Icon>,
          "className": "Header-Left",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <pure(Home) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon>
              <pure(RssFeed) />
            </Icon>,
            <WithStyles(Typography)
              color="inherit"
              variant="subheading"
            >
              +
            </WithStyles(Typography)>,
            <Icon>
              <TiSocialTwitter />
            </Icon>,
            <WithStyles(Typography)
              color="inherit"
              variant="subheading"
            >
              Feed
            </WithStyles(Typography)>,
          ],
          "className": "Header-Center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(RssFeed) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "+",
              "color": "inherit",
              "variant": "subheading",
            },
            "ref": null,
            "rendered": "+",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <TiSocialTwitter />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Feed",
              "color": "inherit",
              "variant": "subheading",
            },
            "ref": null,
            "rendered": "Feed",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Header-Right",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Header-Left"
          >
            <Icon>
              <pure(Home) />
            </Icon>
          </div>,
          <div
            className="Header-Center"
          >
            <Icon>
              <pure(RssFeed) />
            </Icon>
            <WithStyles(Typography)
              color="inherit"
              variant="subheading"
            >
              +
            </WithStyles(Typography)>
            <Icon>
              <TiSocialTwitter />
            </Icon>
            <WithStyles(Typography)
              color="inherit"
              variant="subheading"
            >
              Feed
            </WithStyles(Typography)>
          </div>,
          <div
            className="Header-Right"
          />,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon>
              <pure(Home) />
            </Icon>,
            "className": "Header-Left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(Home) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon>
                <pure(RssFeed) />
              </Icon>,
              <WithStyles(Typography)
                color="inherit"
                variant="subheading"
              >
                +
              </WithStyles(Typography)>,
              <Icon>
                <TiSocialTwitter />
              </Icon>,
              <WithStyles(Typography)
                color="inherit"
                variant="subheading"
              >
                Feed
              </WithStyles(Typography)>,
            ],
            "className": "Header-Center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(RssFeed) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "+",
                "color": "inherit",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "+",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TiSocialTwitter />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Feed",
                "color": "inherit",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "Feed",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Header-Right",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
